cpmpy/expressions/globalconstraints.py

```diff
@@ -289,7 +289,32 @@ def deepcopy(self, memodict={}):
 
     # TODO: value()
 
-# Numeric Global Constraints (with integer-valued return type)
+
+# syntax of the form 'if b then x == 9 else x == 0' is not supported
+# a little helper:
+class IfThenElse(GlobalConstraint):
+    def __init__(self, condition, if_true, if_false):
+        super().__init__("ite", [condition, if_true, if_false], is_bool=True)
+
+    def value(self):
+        condition, if_true, if_false = self.args
+        condition_val = argval(condition)
+        true_val = argval(if_true)
+        false_val = argval(if_false)
+        if condition_val:
+            return true_val
+        else:
+            return false_val
+
+    def decompose(self):
+        condition, if_true, if_false = self.args
+        return (condition.implies(if_true) & \
+                (~condition).implies(if_false)
+                )
+
+    def __repr__(self):
+        condition, if_true, if_false = self.args
+        return "If {} Then {} Else {}".format(condition, if_true, if_false)
```

Comment:
Maybe put brackets around the `if_true` and `if_false`?

Reply:
After discussion, the general policy to put brackets in CPMpy is to do it for non-variable subexpressions. This is probably enough to disambiguate most operators (e.g., `-x+y` vs `-(x+y)`). However, as Wout remarks, the words in ITE work as well as brackets to disambiguate. For `condition` and `if_true` this is clear, for `if_false` there is no "closing" keyword, but as we do not have any post-fix operators (e.g., `if foo then bar else baz**'**`) I think `if_false` is fine too.

Reply:
So yeah, agree with Wout that for now we don't need them.